app/models/language.rb

```diff
@@ -58,55 +58,108 @@ def self.menu_options
   # Get a list of the top N contributors to a language's translations.
   # This is used by the app layout, so must cause mimimal database load.
   def top_contributors(num = 10)
-    user_ids = self.class.connection.select_rows(%(
-      SELECT user_id
-      FROM translation_strings
-      WHERE language_id = #{id} AND user_id != 0
-      GROUP BY user_id
-      ORDER BY COUNT(id) DESC
-      LIMIT #{num}
-    ))
-    if user_ids.any?
-      user_ids = self.class.connection.select_rows(%(
-        SELECT id, login
-        FROM users
-        WHERE id IN (#{user_ids.join(",")})
-        ORDER BY FIND_IN_SET(id, '#{user_ids.join(",")}')
-      ))
-    end
-    user_ids
+    ts = Arel::Table.new(:translation_strings)
+    u = Arel::Table.new(:users)
+    TranslationString.where(language: self).where.not(user_id: 0).
+      group(:user_id).order(ts[:id].count).limit(num).
+      joins(:user).pluck(u[:id], u[:login])
   end
 
   # Count the number of lines the user has translated.  Include edits, as well.
   # It counts paragraphs, actually, and weights them according to length.
   def self.calculate_users_contribution(user)
```

Comment:
This is one of the gnarliest bits of Arel we have in these PRs, but it does produce the desired result. :)

The way I can imagine simplifying it is to somehow do the `GROUP_CONCAT` in Ruby, rather than SQL, but i can't wrap my head around it — i'm not so sure how the scoring actually works. (I know it's counting lines of versions...)

Reply:
ohhhh hold on... there's an Arel Extension for that!

Reply:
Incorporated arel-extension group_concat.